dataactvalidator/filestreaming/loadFile.py

```diff
@@ -1,53 +1,216 @@
+    interface.session.query(model).delete()
+    interface.session.commit()
+
+    data = pd.read_csv(filename, dtype=str)
+    data = LoaderUtils.cleanData(
+        data,
+        model,
+        {"year": "budget_year",
+         "agency_id": "agency_id",
+         "alloc_id": "allocation_transfer_id",
+         "account": "account_number",
+         "pa_code": "program_activity_code",
+         "pa_name": "program_activity_name"},
+        {"program_activity_code": {"pad_to_length": 4},
+         "agency_id": {"pad_to_length": 3},
+         "account_number": {"pad_to_length": 4},
+         "allocation_transfer_id": {"pad_to_length": 3}}
+    )
+    # because we're only loading a subset of program activity info,
+    # there will be duplicate records in the dataframe. this is ok,
+    # but need to de-duped before the db load.
+    data.drop_duplicates(inplace=True)
+    # insert to db
+    LoaderUtils.insertDataframe(data, model.__table__.name, interface.engine)
+
+
+def loadSF133(filename, fiscal_year, fiscal_period, force_load=False):
+    interface = ValidatorValidationInterface()
+    model = SF133
+
+    existing_records = interface.session.query(model).filter(
+        and_(model.fiscal_year == fiscal_year, model.period == fiscal_period))
+    if force_load:
+        # force a reload of this period's current data
+        print('Force SF 133 load: deleting existing records for {} {}'.format(
+            fiscal_year, fiscal_period))
+        existing_records.delete()
+        interface.session.commit()
+    elif existing_records.count():
+        # if there's existing data & we're not forcing a load, skip
+        print('SF133 {} {} already in database ({} records). Skipping file.'.format(
+            fiscal_year, fiscal_period, existing_records.count()))
+        return
+
+    data = pd.read_csv(filename, dtype=str)
+    data = LoaderUtils.cleanData(
+        data,
+        model,
+        {"ata": "allocation_transfer_agency",
+         "aid": "agency_identifier",
+         "availability_type_code": "availability_type_code",
+         "bpoa": "beginning_period_of_availa",
+         "epoa": "ending_period_of_availabil",
+         "main_account": "main_account_code",
+         "sub_account": "sub_account_code",
+         "fiscal_year": "fiscal_year",
+         "period": "period",
+         "line_num": "line",
+         "amount_summed":
+        "amount"},
+        {"allocation_transfer_agency": {"pad_to_length": 3},
+         "agency_identifier": {"pad_to_length": 3},
+         "main_account_code": {"pad_to_length": 4},
+         "sub_account_code": {"pad_to_length": 3},
+         "amount": {"strip_commas": True}}
+    )
+    # todo: find out how to handle dup rows (e.g., same tas/period/line number)
+    # line numbers 2002 and 2012 are the only duped line numbers,
+    # and they are not used by the validation rules, so for now
+    # just remove them before loading our SF-133 table
+    dupe_line_numbers = ['2002', '2102']
```

Comment:
Thanks for the comments! I don't understand them yet, but they're much better than having magic numbers